globus_automate_client/cli/flows.py

```diff
@@ -76,11 +103,11 @@ def flow_deploy(
     definition: str = typer.Option(
         ...,
         help=(
-            "JSON representation of the Flow to deploy. May be provided as a filename "
-            "or a raw JSON string."
+            "JSON or YAML representation of the Flow to deploy. May be provided as a filename "
+            "or a raw string representing a JSON object or YAML definition."
         ),
         prompt=True,
-        callback=json_validator_callback,
+        callback=input_validator_callback,
```

Comment:
Looks like this is headed toward what I was thinking above, this'll do the file management stuff and then the parsing can assume it has the string representation.

Probably means the file open/reading stuff can be removed from the json/yaml-specific validators?

Reply:
Yes this is what I was thinking. Parse the input (string or file) then return a string to be parsed as the appropriate type (json, yaml)